opm/material/fluidsystems/Co2BrineFluidSystem.hh

```diff
@@ -0,0 +1,190 @@
+        *
+        * \copydetails Doxygen::compIdxParam
+        */
+        static Scalar criticalVolume(unsigned compIdx)
+        {
+            switch (compIdx) {
+                case Comp0Idx: return Comp0::criticalVolume();
+                case Comp1Idx: return Comp1::criticalVolume();
+                default: throw std::runtime_error("Illegal component index for criticalVolume");
+            }
+        }
+
+        //! \copydoc BaseFluidSystem::molarMass
+        static Scalar molarMass(unsigned compIdx)
+        {
+            switch (compIdx) {
+                case Comp0Idx: return Comp0::molarMass();
+                case Comp1Idx: return Comp1::molarMass();
+                default: throw std::runtime_error("Illegal component index for molarMass");
+            }
+        }
+
+        /*!
+         * \brief Returns the interaction coefficient for two components.
+         *.
+         */
+        static Scalar interactionCoefficient(unsigned /*comp1Idx*/, unsigned /*comp2Idx*/)
+        {
+            return 0.0;
+        }
+
+        //! \copydoc BaseFluidSystem::phaseName
+        static const char* phaseName(unsigned phaseIdx)
+        {
+                static const char* name[] = {"o",  // oleic phase
+                                             "g"};  // gas phase
+
+                assert(0 <= phaseIdx && phaseIdx < 2);
+                return name[phaseIdx];
+        }
+
+        //! \copydoc BaseFluidSystem::componentName
+        static const char* componentName(unsigned compIdx)
+        {
+                static const char* name[] = {
+                        Comp0::name(),
+                        Comp1::name(),
+                };
+
+                assert(0 <= compIdx && compIdx < 3);
+                return name[compIdx];
+        }
+
+        /*!
+         * \copydoc BaseFluidSystem::density
+         */
+        template <class FluidState, class LhsEval = typename FluidState::Scalar, class ParamCacheEval = LhsEval>
+        static LhsEval density(const FluidState& fluidState,
+                               const ParameterCache<ParamCacheEval>& paramCache,
+                               unsigned phaseIdx)
+        {
+
+            LhsEval dens;
+            if (phaseIdx == oilPhaseIdx || phaseIdx == gasPhaseIdx) {
+                dens = fluidState.averageMolarMass(phaseIdx) / paramCache.molarVolume(phaseIdx);
+            }
+            return dens;
+
+        }
+
+        /*!
+         * \copydoc BaseFluidSystem::viscosity
+         */
+        template <class FluidState, class LhsEval = typename FluidState::Scalar, class ParamCacheEval = LhsEval>
+        static LhsEval viscosity(const FluidState& fluidState,
+                                 const ParameterCache<ParamCacheEval>& paramCache,
+                                 unsigned phaseIdx)
+        {
+            // Use LBC method to calculate viscosity
+            LhsEval mu;
+            mu = ViscosityModel::LBC(fluidState, paramCache, phaseIdx);
```

Comment:
Following the discussion before, I think here the `LBCModified` should be used, right?

Reply:
Both can be used, when comparing to the Julia code the standard LBC is used

Reply:
Currently, LBCCo2Rich is not used anywhere. Maybe you can set a flag to `Co2BrineFluidSystem` to control whether `LBC` or `LBCCo2Rich` is used for the viscosity calcualtion?